galaxy/worker/tasks.py

```diff
@@ -349,6 +350,20 @@ def _update_quality_score(import_task):
     LOG.debug(u'repo quality score: {}'.format(repository.quality_score))
 
 
+def _remove_last_task_msg(import_task):
+    last_task = models.ImportTask.objects.filter(
```

Comment:
Probably it makes more sense not limiting cleanup by only last import task and just cleanup all import tasks excluding current task.